packages/api/src/submittable/createClass.ts

```diff
@@ -184,7 +184,14 @@ export function createClass <ApiType extends ApiTypes> ({ api, apiType, decorate
     }
 
     #makeSignOptions = (options: Partial<SignerOptions>, extras: { blockHash?: Hash; era?: ExtrinsicEra; nonce?: Index }): SignatureOptions => {
+      const assetId = options.assetId
+        ? this.registry.createType('Option<AssetId>', options.assetId)
+        : this.registry.createType('Option<AssetId>');
+
+      delete options.assetId;
+
```

Comment:
We should be able to get rid of this with the adjustments to AnyNumber commented on. It will just convert at the point of use.